core/src/test/java/com/linecorp/armeria/server/RequestIdHeadersTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.BlockingWebClient;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.RequestId;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+import com.linecorp.armeria.testing.server.ServiceRequestContextCaptor;
+
+class RequestIdHeadersTest {
+
+    static final String REQUEST_ID_KEY = "X-RequestId";
+    static final String REQUEST_ID_VAL = "From_Client";
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.requestIdGenerator((ctx) -> {
+                  final String header = ctx.headers().get(REQUEST_ID_KEY, "default");
+
+                  if (header.equals(REQUEST_ID_VAL)) {
+                      return RequestId.of(123L);
+                  }
+
+                  return RequestId.random();
+              })
```

Comment:
What do you think of just using the header value as the request id? I think this can simplify the test logic a little

```suggestion
            sb.requestIdGenerator(ctx -> RequestId.of(Long.parseLong(ctx.headers().get(REQUEST_ID_KEY))))
```

Reply:
ctx.headers().get(REQUEST_ID_KEY) can produce null and It can produce NumberFormatException or any Exception

So is it be okay for try catch for this configure method?

like below
```
    static ServerExtension server = new ServerExtension() {
        @Override
        protected void configure(ServerBuilder sb) throws Exception {
            sb.requestIdGenerator((ctx) -> {
                  try {
                      return RequestId.of(
                              Long.parseLong(Objects.requireNonNull(ctx.headers().get(REQUEST_ID_KEY))));
                  } catch (Exception e) {
                      return RequestId.random();
                  }
              })
              .service("/", (ctx, req) -> HttpResponse.of(200));
        }
    };
```

Reply:
If we do exception handling as suggested by https://github.com/line/armeria/pull/4691#discussion_r1116571214, I thought the result would be the same as your suggestion 😄